features/consent/src/main/java/in/projecteka/jataayu/consent/ui/fragment/RequestedConsentDetailsFragment.kt

```diff
@@ -232,4 +243,26 @@ class RequestedConsentDetailsFragment : BaseFragment(), ItemClickCallback,
             }
         }
     }
+
+    fun validateUser() {
+        if (context?.getPinCreated()!!) {
+            val intent = Intent(context, PinVerificationActivity::class.java)
+            startActivityForResult(intent, 301)
+        } else {
+            val intent = Intent(context, CreatePinActivity::class.java)
+            startActivityForResult(intent, 201)
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode == Activity.RESULT_OK){
+            if (requestCode == 201) {
+                activity?.setPinCreated(true)
```

Comment:
this should be done once the pin is created. Why require a result for this. As far as this fragment is concerned, it is not responsible for knowing whether or not a pin is needed/not needed. It should just know whether or not user was verified.